common/script/api-v3/scoreTask.js

```diff
@@ -0,0 +1,251 @@
+
+  let hpMod = delta * conBonus * task.priority * 2; // constant 2 multiplier for better results
+  stats.hp += Math.round(hpMod * 10) / 10; // round to 1dp
+  return stats.hp;
+}
+
+function _addPoints (user, task, stats, direction, delta) {
+  // ===== CRITICAL HITS =====
+  // allow critical hit only when checking off a task, not when unchecking it:
+  let _crit = delta > 0 ? user.fns.crit() : 1;
+  // if there was a crit, alert the user via notification
+  if (_crit > 1) user._tmp.crit = _crit;
+
+  // Exp Modifier
+  // ===== Intelligence =====
+  // TODO Increases Experience gain by .2% per point.
+  let intBonus = 1 + user._statsComputed.int * 0.025;
+  stats.exp += Math.round(delta * intBonus * task.priority * _crit * 6);
+
+  // GP modifier
+  // ===== PERCEPTION =====
+  // TODO Increases Gold gained from tasks by .3% per point.
+  let perBonus = 1 + user._statsComputed.per * 0.02;
+  let gpMod = delta * task.priority * _crit * perBonus;
+
+  if (task.streak) {
+    let currStreak = direction === 'down' ? task.streak - 1 : task.streak;
+    let streakBonus = currStreak / 100 + 1; // eg, 1-day streak is 1.01, 2-day is 1.02, etc
+    let afterStreak = gpMod * streakBonus;
+    if (currStreak > 0 && gpMod > 0) {
+      user._tmp.streakBonus = afterStreak - gpMod; // keep this on-hand for later, so we can notify streak-bonus
+    }
+
+    stats.gp += afterStreak;
+  } else {
+    stats.gp += gpMod;
+  }
+}
+
+function _changeTaskValue (user, task, direction, times, cron) {
+  let addToDelta = 0;
+
+  // If multiple days have passed, multiply times days missed
+  _.times(times, () => {
+    // Each iteration calculate the nextDelta, which is then accumulated in the total delta.
+    let nextDelta = !cron && direction === 'down' ? _calculateReverseDelta(task, direction) : _calculateDelta(task, direction, cron);
+
+    if (task.type !== 'reward') {
+      if (user.preferences.automaticAllocation === true && user.preferences.allocationMode === 'taskbased' && !(task.type === 'todo' && direction === 'down')) {
+        user.stats.training[task.attribute] += nextDelta;
+      }
+
+      if (direction === 'up') { // Make progress on quest based on STR
+        user.party.quest.progress.up = user.party.quest.progress.up || 0;
+
+        if (task.type === 'todo' || task.type === 'daily') {
+          user.party.quest.progress.up += nextDelta * (1 + user._statsComputed.str / 200);
+        } else if (task.type === 'habit') {
+          user.party.quest.progress.up += nextDelta * (0.5 + user._statsComputed.str / 400);
+        }
+      }
+
+      task.value += nextDelta;
+    }
+
+    addToDelta += nextDelta;
+  });
+
+  return addToDelta;
+}
+
+export default function scoreTask (options = {}, req = {}) {
+  let {user, task, direction, times = 1, cron = false} = options;
+  let delta = 0;
+  let stats = {
+    gp: user.stats.gp,
+    hp: user.stats.hp,
+    exp: user.stats.exp,
+  };
+
+  // TODO return or pass to cb, don't add to user object
```

Comment:
Whatever we do, we need some way to pass it on to the API client in the response.